### DISCUSSION TITLE
Importing dsl files in another dsl file

### DISCUSSION THREAD

**Discussion by SamuelTsokwa** [eclipse-langium/langium#914] [Q&A] [answered]
This might be a dumb question.
I'm currently developing a DSL for some frontend work. As a part of this, I would like to have some components that come pre-packaged with the language. For example, A user of the DSL should be able to reuse a Text component that I have pre-defined.
Assuming I keep my pre-defined components in one file. Is there a way to import/expose those components in a different file?
For example:
// Components.mydsl

`export component Text {
      .... some text defintion
}`

//someDifferentFile.mydsl
`import  Text`
`.......`
`......`
` Text()`

**Comment by luan-xiaokun** [comment, marked as answer]
I have not practiced it myself, but the [builtin libraries](https://langium.org/guides/builtin-library/) guide seems like a good place to start. You may also want to read more about how cross-reference are resolved in [document lifecycle](https://langium.org/docs/document-lifecycle/)'s document.

If I understand your question correctly, #853 seems to be a perfect answer.

**Reply by SamuelTsokwa** [reply]
Thanks! I'll look into it

**Comment by goto40** [comment]
See also #458 . This may clarify how to "import" and reference elements from other DSL files.

**Comment by goto40** [comment]
There is [one example included in langium](https://github.com/langium/langium/tree/main/examples/requirements/example), where  two DSLs are combined and elements from one DSL reference elements from the other DSL (w/o import statements, using global visibility).
